atlasdb-api/src/main/java/com/palantir/atlasdb/spi/AtlasDbFactory.java

```diff
@@ -38,6 +38,13 @@
 
     String getType();
 
+    default KeyValueServiceConfig createMergedKeyValueServiceConfig(
```

Comment:
```suggestion
    default KeyValueServiceConfig createKeyValueServiceConfig(
```

Reply:
I'd also suggest we change the field name, etc. That fact that it's merged isn't really relevant to consumers - there should be a single config that is the source of truth.

Reply:
I'd rather differentiate the names